### PR TITLE
Add anonymous envelope (zk) verification with test

### PR DESCRIPTION
**Pull request by arnaucube**

### CODE REVIEW

File: vochain/transaction.go
Comment (arnaucube) [inline vochain/transaction.go:218]: This will be done in a future PR.

File: vochain/transaction.go
Comment (ed255) [inline vochain/transaction.go]: check that `proofZkSNARK.CircuitParametersIndex < len(app.ZkVks)` to avoid an easy panic.

File: vochain/transaction.go
Comment (ed255) [inline vochain/transaction.go]: This should never happen because app.ZkVks is  slice, and we should never add a nil pointer to that slice.

**Review by ed255 — CHANGES_REQUESTED** [review]

File: vochain/transaction.go
Comment (arnaucube) [reply vochain/transaction.go]: Right! Forgot about it when finishing the PR, updated

**Review by ed255 — APPROVED** [review]
Please, wait until this is merged https://github.com/vocdoni/vocdoni-node/pull/260 to merge this PR to resolve possible conflicts in chronological order :)

**Comment by arnaucube** [comment]
Rebased to `master` branch, including the changes from https://github.com/vocdoni/vocdoni-node/pull/260 .

**Review by mvdan — APPROVED** [review]

**Pull request merged**
